dev/PushNotifications/PushNotificationReceivedEventArgs.cpp

```diff
@@ -23,49 +23,108 @@ namespace winrt
-        return { rawNotificationData, rawNotificationData + (m_rawNotification.Length() * sizeof(uint8_t)) };
+        return { m_rawNotification, m_rawNotification + (m_length * sizeof(uint8_t)) };
     }
 
     winrt::BackgroundTaskDeferral PushNotificationReceivedEventArgs::GetDeferral()
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
-
-        return m_backgroundTaskInstance.GetDeferral();
+        if (m_isBIAvailable)
+        {
+            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
+
+            return m_backgroundTaskInstance.GetDeferral();
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
     winrt::event_token PushNotificationReceivedEventArgs::Canceled(winrt::BackgroundTaskCanceledEventHandler const& handler)
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
-
-        return m_backgroundTaskInstance.Canceled(handler);
+        if (m_isBIAvailable)
+        {
+            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
+
+            return m_backgroundTaskInstance.Canceled(handler);
+        }
+        else
+        {
+            return { 0 };
+        }
     }
 
     void PushNotificationReceivedEventArgs::Canceled(winrt::event_token const& token) noexcept
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
+        if (m_isBIAvailable)
+        {
+            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
 
-        m_backgroundTaskInstance.Canceled(token);
+            m_backgroundTaskInstance.Canceled(token);
+        }
     }
 
     bool PushNotificationReceivedEventArgs::Handled()
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_args, "Background activation cannot call this.");
-
-        return m_args.Cancel();
+        if (m_isBIAvailable)
+        {
+            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_args, "Background activation cannot call this.");
+
+            return m_args.Cancel();
+        }
+        else
+        {
+            return m_handledUnpackaged;
+        }
     }
 
     void PushNotificationReceivedEventArgs::Handled(bool value)
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_args, "Background activation cannot call this.");
-
-        m_args.Cancel(value);
+        if (m_isBIAvailable)
+        {
+            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_args, "Background activation cannot call this.");
+
+            m_args.Cancel(value);
+        }
+        else
+        {
+            if (value)
+            {
+                wchar_t processName[1024];
```

Comment:
Can this code (the get module file name) be deduplicated with the above as well?

Reply:
Added GetCurrentProcessPath in externs.h